assets/components/src/plugin-toggle/index.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Component, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import { ActionCard, Waiting } from '../';
+import './style.scss';
+
+/**
+ * Plugin toggle group component.
+ */
+class PluginToggle extends Component {
+	state = {
+		pluginInfo: {},
+	};
+
+	componentDidMount = () => {
+		this.retrievePluginInfo();
+	};
+
+	/**
+	 * Retrieve complete data about Newspack plugins.
+	 */
+	retrievePluginInfo = () => {
+		const { plugins } = this.props;
+		return new Promise( ( resolve, reject ) => {
+			apiFetch( { path: '/newspack/v1/plugins/' } ).then( pluginInfo =>
+				this.setState( { pluginInfo } )
+			);
+		} );
+	};
+
+	/**
+	 * Install/activate or remove a plugin.
+	 */
+	managePlugin = ( plugin, value ) => {
+		const { plugins } = this.props;
+		const { pluginInfo } = this.state;
+		const action = value ? 'configure' : 'uninstall';
```

Comment:
Uninstalling the plugin when toggled off seems drastic. Why uninstall instead of deactivate?

Reply:
Great idea: https://github.com/Automattic/newspack-plugin/pull/406/commits/ec0567bea3dcb6a901024a68dbdd78db7f973176